notifications_utils/recipients.py

```diff
@@ -276,6 +276,25 @@ def rows_with_missing_data(self):
     def rows_with_message_too_long(self):
         return self._filter_rows("message_too_long")
 
+    @property
+    def rows_with_combined_variable_content_too_long(self):
+        """
+        Checks if the length of all variable, plus the length of the template content,
+        exceeds the SMS limit of 612 characters. Counts non-GSM characters as 2.
+        """
+        rows_too_long = []
+        if self.rows_as_list:
+            for row in self.rows_as_list:
+                if row.personalisation and self.template and self.template.placeholders:
+                    variable_length = 0
+                    for variable in row.personalisation.as_dict_with_keys(self.template.placeholders).values():
+                        if variable:
+                            variable_length += sum(1 if char in SanitiseSMS.ALLOWED_CHARACTERS else 2 for char in str(variable))
+                    total_length = variable_length + (len(self.template.content) if self.template else 0)
+                    if total_length > SMS_CHAR_COUNT_LIMIT:
+                        rows_too_long.append(row)
+        return (row for row in rows_too_long)
```

Comment:
why are we doing this?

Reply:
The intention is to return a `generator` rather than a `list` to remain consistent with the other properties that return rows meeting certain criteria. E.g. [`rows_with_missing_data`](https://github.com/cds-snc/notification-utils/pull/278/files/3ffd44c37e17148463582049b10cf00b760433aa#diff-a8a994bf655634f89dc7439880708b4ff0d780ac1bd8033827d8aaa2692a8e0fR272).

That said this should be refactored to yield the rows in the loops instead of storing them, as that's the whole point of using generators.🤦🏻‍♂️

Reply:
Updated, let me know if this makes more sense!